geostore/check_files_checksums/utils.py

```diff
@@ -75,30 +74,38 @@ def validate(self, hash_key: str, range_key: str) -> None:
         else:
             self.logger.info(LOG_MESSAGE_VALIDATION_COMPLETE, extra={"outcome": Outcome.PASSED})
             self.validation_result_factory.save(item.url, Check.CHECKSUM, ValidationResult.PASSED)
+            if not file_in_staging:
+                item.delete()
```

Comment:
This doesn't seem right. The file is mentioned in the staging metadata, so it should stay unchanged.